Tests/UnitTests/FileShareClientTests/BatchAttributeSearchTests.cs

```diff
@@ -152,13 +152,43 @@ public void TestNameOfAndAttributeValuesinToStringMethod()
             var colourList = new List<string> { "red", "blue" };
 
             var searchBatchAttributes = new BatchAttributesSearchAttribute
-                    {
-                        Key = "Colour",Values = colourList
+            {
+                Key = "Colour",
+                Values = colourList
             };
             var attributeValues = searchBatchAttributes.ToString();
             Assert.AreEqual("class BatchAttributesSearchAttribute {\n Key: Colour\n Values: red, blue\n}\n", attributeValues);
         }
 
+        #region BatchSearch with MaxAttributeValueCount
+
+        [Test]
+        public async Task DoesBatchAttributeSearchReturnsSucessWithMaxAttributeValueCountandFilter()
+        {
+            int MaxAttributeValueCount = 2;
+
+            var response = await fileShareApiClient.BatchAttributeSearch("$batch(key) eq 'value'", MaxAttributeValueCount,cancellationToken: CancellationToken.None);
+            Assert.AreEqual("/basePath/attributes/search", lastRequestUri.AbsolutePath);
+            Assert.AreEqual("?$filter=$batch(key)%20eq%20%27value%27&maxAttributeValueCount=2", lastRequestUri.Query);
+            Assert.IsTrue(response.IsSuccess);
+
+        }
+        [Test]
+        public async Task DoesBatchAttributeSearchReturnsBadRequestWithMaxAttributeValueCountZeroandFilter()
```

Comment:
Please delete test `DoesBatchAttributeSearchReturnsBadRequestWithMaxAttributeValueCountZeroandFilter` because it's only testing the behaviour of the mock http client factory

Reply:
Changed my mind, please leave it in